### PR TITLE
Feat/15669195 activity feed table

### PR DESCRIPTION
**Pull request by TomaszChmielPagepro**
## Description

This PR contains changes from https://github.com/JoinColony/colonyCDapp/pull/1336 as this hook was needed for table implementation. The status column is not sortable for now as `useActivityFeed` does not support this.

**New stuff** ✨

Added colony activity table without filters as this PR has gotten pretty big already.

image
image

### CODE REVIEW

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @TomaszChmielPagepro! Amazing work on this and great to see it come to life.

A few changes to align it with the design.

1. When I select the date arrow, it hides (I'm guessing while the data is being pulled) and then comes back again only when I try to select again. Can it show the whole time to avoid user confusion? 

Animation111

2. The 'forced' pill should be replaced with the passed version? I understand this isn't aligned with the design. 

image

Link to Figma if you need the component - https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=16368-84713&mode=design&t=qE26uhUdRpPwHXpn-4

3. Background hover on each row is missing (you can find the correct example on the incoming funds table). 

4. When a table has less than 10 results, the padding allowed for pagination and number of pages should be removed and matched with the design in Figma to remove the empty space at the bottom of the table. 

image

Link to Figma for reference: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=8039-377316&mode=design&t=qE26uhUdRpPwHXpn-4

5. The title of each action/motion should match Figma:
image

6. There's an issue when switching between actions:
switching-between-actions

7. The share button in the dropdown menu when selected, should use the same copy tooltip that's in the dashboard PR on the share icon and also in the dropdown menu - https://github.com/JoinColony/colonyCDapp/pull/1315

image

8. The View on Ethscan button should use the logic from this component src/components/shared/TransactionLink/TransactionLink.tsx

9. The status pills are not being picked up correctly. I made a motion but it's showing forced. 

10. Active action is not appearing in the list:
image

11. Time is showing as a negative value
image

12. When there are more than 10 items, the pagination shows but the Page # is missing bottom left.
image

13. When the motion is in the finalizable state, there is no timer. Here you can see the timer showing incorrectly.
image

14. I noticed this error in the console:
image

15. When I scroll, there is no padding at the bottom of the page. Globally, there should be 
16px in the content to match the bottom of the left nav container.
image

16. I'm seeing double scrollbars on the right
image

**Review by rdig — CHANGES_REQUESTED** [review]
- Shouldn't this PR based on `feat/motion-status-filtering` and not `master`, seeing as it's based on #1336 ? Otherwise, merging this will be quite problematic

- Longer domain names break the UI
Screenshot from 2023-11-23 13-04-36

- The action with `zzz` as a title is a Motion, the other one is a plain action. There should be some distinction between them
Screenshot from 2023-11-23 13-04-53

**Comment by TomaszChmielPagepro** [comment]
@rdig 
1. This is not the case anymore as `feat/motion-status-filtering` has been merged to master. I have rebased my branch on master as well.
2. Elipsis added
3. @arrenv told me that you guys have discussed this internally and it's not an issue.

**Comment by TomaszChmielPagepro** [comment]
Thanks for the review @melyndav 

1. I have disabled clearing sort state
2. Forced changed to Passed
3. Background added
4. Removed empty space
5. Font style updated
6. There is a related PR for action sidebar closing which may have an impact on this. I have added some cases for initial values loading in `src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts` but it needs more fixes as not every action is handled properly and I don't want to make this PR too big
7. I will fix this when the PR mentioned by you is merged because it seems like some logic may be shared here (or if it will take too long to merge this PR I will just do it here anyway tomorrow)
8. Used `TransactionLink` component
9. I couldn't reproduce that. Can you please share more details here?
10. I have added refetch on action creation but we may need to refetch the list on motion state changes as well. I will try to add this tomorrow.
11. Fixed
12. We don't have information about the total page count. Do you want me to add the current page here only?
13. When the motion state is `Finalizable` the timer will show negative time as there is no timer value for this state in the API. Am I missing something here?
image
14. Fixed
15. Padding added
16. Is there any interaction that could cause this? I can't reproduce this

**Comment by melyndav** [comment]
Hey @TomaszChmielPagepro! To answer some of your questions - 

> 9. I couldn't reproduce that. Can you please share more details here?
Leave this for now and we'll see if we can pick this up with further testing. 

> 12. We don't have information about the total page count. Do you want me to add the current page here only?
Yes please. So it will display Page 1 for example.

> 13. When the motion state is `Finalizable` the timer will show negative time as there is no timer value for this state in the API. Am I missing something here?
Can we hide it when it's a negative value?

> 16. Is there any interaction that could cause this? I can't reproduce this
I couldn't replicate this again. I'll keep on eye it.

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @TomaszChmielPagepro super work on the updates I requested, so thank you!

Just a few more - 

1. The pagination (page #) shouldn't show when there is 10 or less results. And the extra padding at the bottom should be removed like Figma: https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=8039-377316&mode=design&t=qE26uhUdRpPwHXpn-4

2. Row heights should be 58px to keep it consistent - 
image

3. For icons, please use the phosphor icon library - 

image
 
 4. The share menu item in each row's dropdown is missing the tooltip copied state as mentioned previously:
 
_The share button in the dropdown menu when selected, should use the same copy tooltip that's in the dashboard PR on the share icon and also in the dropdown menu_ - https://github.com/JoinColony/colonyCDapp/pull/1315

image

5. The View on Ethscan button should use the logic from this component (as mentioned in previous comment) 
src/components/shared/TransactionLink/TransactionLink.tsx

File: src/components/v5/common/ActionSidebar/partials/forms/AdvancedPaymentForm/utils.tsx [hunk 1]
```diff
@@ -28,35 +29,30 @@ const getTokensText = (
...
   switch (tokensCount) {
     case 0:
-      return 'multiple tokens';
+      return 'with multiple tokens';
     case 1:
-      return '1 token';
+      return 'with 1 token';
     default:
-      return `${tokensCount} tokens`;
+      return `with ${tokensCount} tokens`;
```
Comment (arrenv) [inline src/components/v5/common/ActionSidebar/partials/forms/AdvancedPaymentForm/utils.tsx:36]: We should be using translation here, right?

File: src/components/v5/common/ActionSidebar/partials/forms/MintTokenForm/utils.tsx [hunk 2]
```diff
@@ -1,83 +1,32 @@
...
+      amount: amount?.amount
+        ? moveDecimal(
+            amount.amount.toString(),
+            getTokenDecimalsWithFallback(colony?.nativeToken?.decimals),
+          )
+        : undefined,
+    },
+    {
+      [ActionTitleMessageKeys.Amount]: '',
+      [ActionTitleMessageKeys.TokenSymbol]: 'native tokens',
```
Comment (arrenv) [inline src/components/v5/common/ActionSidebar/partials/forms/MintTokenForm/utils.tsx:29]: Translation issue here as well.

File: src/components/v5/common/ActionSidebar/partials/forms/SplitPaymentForm/utils.tsx [hunk 3]
```diff
@@ -1,88 +1,41 @@
...
+            amount.amount.toString(),
+            getTokenDecimalsWithFallback(token?.decimals),
+          )
+        : undefined,
+      token,
+    },
+    {
+      [ActionTitleMessageKeys.Recipient]: '',
+      [ActionTitleMessageKeys.Amount]: 'an unspecified amount of',
+      [ActionTitleMessageKeys.TokenSymbol]: '',
```
Comment (arrenv) [inline src/components/v5/common/ActionSidebar/partials/forms/SplitPaymentForm/utils.tsx:38]: Translation.

File: src/utils/colonyActions.ts [hunk 4]
```diff
@@ -535,66 +536,36 @@ export const normalizeRolesForAction = (
...
-      roleTitle += roleTitle ? ' and remove the' : 'Remove the';
-      roleTitle += getFormattedRoleList(unassignedRoles, null);
-    }
+  const normalizedRoles = normalizeRolesForAction(roles);
+  const assignedRoles = normalizedRoles.filter((role) => role.setTo);
 
-    roleTitle += normalizedRoles.length > 1 ? ' permissions' : ' permission';
+  if (isEmpty(assignedRoles)) {
+    return {
+      direction: 'Remove',
```
Comment (arrenv) [inline src/utils/colonyActions.ts:559]: Translation.

File: src/utils/colonyActions.ts [hunk 5]
```diff
@@ -535,66 +536,36 @@ export const normalizeRolesForAction = (
...
+      roleTitle: '',
+    };
   }
 
+  const role = getRole(assignedRoles.map(({ id }) => id));
+
   return {
-    roleTitle,
-    direction,
+    direction: 'Assign',
```
Comment (arrenv) [inline src/utils/colonyActions.ts:567]: Translation.

**Review by arrenv — CHANGES_REQUESTED** [review]
Great work on this @TomaszChmielPagepro, I picked up a couple of missing translations.

**Comment by TomaszChmielPagepro** [comment]
@melyndav I have fixed your comments

**Comment by TomaszChmielPagepro** [comment]
@arrenv Translations added

File: src/images/icons/arrow-down.svg
Comment (arrenv) [inline src/images/icons/arrow-down.svg:1]: This can be removed now, right?

**Review by arrenv — CHANGES_REQUESTED** [review]
Thank you @TomaszChmielPagepro. These values should be formatted.

image

**Review by melyndav — APPROVED** [review]
Approved from my side @TomaszChmielPagepro great work on this. Love the skeleton loader! Thanks for all your work :)

File: src/images/icons/arrow-down.svg
Comment (TomaszChmielPagepro) [reply src/images/icons/arrow-down.svg:1]: image
image
 Seems like it's still used

**Review by melyndav — DISMISSED** [review]

**Review by arrenv — APPROVED** [review]
Huge amount of work here, everything I tested showed it working as expected. Very nice work! Thank you @TomaszChmielPagepro.

File: src/components/common/ColonyActions/ActionDetailsPage/DefaultAction/DefaultActionContent.tsx [hunk 6]
```diff
@@ -15,6 +15,7 @@ import TimeRelative from '~shared/TimeRelative/TimeRelative';
 import ActionAnnotation from './ActionAnnotation';
 
 import styles from './DefaultActionContent.css';
+import { useNetworkInverseFee } from '~hooks';
```
Comment (ArmandoGraterol) [inline src/components/common/ColonyActions/ActionDetailsPage/DefaultAction/DefaultActionContent.tsx:18]: This import should be grouped with the other relative imports

File: src/utils/colonyActions.ts [hunk 7]
```diff
@@ -17,6 +17,7 @@ import { ColonyActionRoles, ColonyActionType } from '~gql';
 import { formatText } from './intl';
 import { MotionVote } from './colonyMotions';
 import { parseSafeTransactionType } from './safes';
+import { getRole } from '~constants/permissions';
```
Comment (ArmandoGraterol) [inline src/utils/colonyActions.ts:20]: Same here

File: src/components/v5/common/ActionSidebar/hooks/useActionDescriptionMetadata.tsx [hunk 8]
```diff
@@ -22,6 +26,10 @@ import { enterRecoveryModeDescriptionMetadataGetter } from '../partials/forms/En
 import { createDecisionDescriptionMetadataGetter } from '../partials/forms/CreateDecisionForm/utils';
 import { managePermissionsDescriptionMetadataGetter } from '../partials/forms/ManagePermissionsForm/utils';
 import { manageColonyObjectivesDescriptionMetadataGetter } from '../partials/forms/ManageColonyObjectivesForm/utils';
+import { ACTION_TYPE_FIELD_NAME } from '../consts';
+import { DescriptionMetadataGetter } from '../types';
+import { ActionTitleMessageKeys } from '~common/ColonyActions/helpers/getActionTitleValues';
+import UserPopover from '~v5/shared/UserPopover';
```
Comment (ArmandoGraterol) [inline src/components/v5/common/ActionSidebar/hooks/useActionDescriptionMetadata.tsx:32]: Same here

**Review by ArmandoGraterol — DISMISSED** [review]
Good work! This is a huge one.

**Comment by TomaszChmielPagepro** [comment]
@ArmandoGraterol import order updated
@rdig Please review your requested changes

**Review by rdig — CHANGES_REQUESTED** [review]
- Motions that don't have at least 10% support should not show up in the activity list

Screenshot from 2023-11-26 04-12-04

Screenshot from 2023-11-26 04-12-12

- Please use the correct explorer name / link contextual the current chain. There's util method that generates them for you.
Screenshot from 2023-11-26 04-13-16
Screenshot from 2023-11-26 04-15-43

**Comment by TomaszChmielPagepro** [comment]
@rdig 
1. I forgot to reenable filter for this, fixed
2. It's done like this. It depends on your env variables which network appears here (this is done the same way as it was in old UI).
image
image

**Review by rdig — CHANGES_REQUESTED** [review]
After the latest changes only 6 actions show, when in fact I have over 100 created at this point:

Screenshot from 2023-11-27 15-23-37
Screenshot from 2023-11-27 15-26-32

- "View on Etherscan" still shows up, despite me being on `ganache`

Screenshot from 2023-11-27 15-25-03
Screenshot from 2023-11-27 15-25-08
Screenshot from 2023-11-27 15-25-14

**Comment by TomaszChmielPagepro** [comment]
1. I don't know why this happens, in this PR I am just consuming `useActivityFeed` hook and in the latest changes I have just brought back `showInActionsList` filter
2. Fixed

**Review by arrenv — APPROVED** [review]
All actions are showing up on my end and "Noexplorer" is showing as expected. Thank you @TomaszChmielPagepro

**Pull request merged**